external/fv3fit/tests/test_io.py

```diff
@@ -29,6 +29,21 @@ class Mock2:
             pass
 
 
+def test_registering_subclass_get_name():
+    register = _Register()
+
+    @register("mock")
+    class Mock:
+        pass
+
+    @register("mock-subclass")
+    class MockSubclass:
```

Comment:
`class MockSubclass(Mock)`?

The case I worked through was that there are two registered subclasses of `Mock` , `MockSubclassA` and `MockSubclassB`, would it return the name for `MockSubclassA` if asked? (It would)

I figured that would be a hairy debug process if there was weirdness with the loop logic and registering multiple subclasses of the same parent.

Reply:
Ha good catch